expr/pool.go

```diff
@@ -12,3 +16,17 @@ var pointSlicePool *sync.Pool
 func Pool(p *sync.Pool) {
 	pointSlicePool = p
 }
+
+// pointSlicePoolGet returns a pointslice of at least minCap capacity.
+// similar code lives also in api.Fix(). at some point we should really clean up our pool code.
```

Comment:
note that the pool is created in the api package and passed into the expr package.
the original idea was to make the expr library reusable in different software, hence the need to pass the pool into it.

But it's starting to make more sense to just have one global pool singleton with a couple methods and directly access that everywhere. Out of scope for this PR though